src/core/thread/mle.cpp

```diff
@@ -4304,14 +4304,22 @@ Error Mle::DetachGracefully(otDetachGracefullyCallback aCallback, void *aContext
     mDetachGracefullyCallback = aCallback;
     mDetachGracefullyContext  = aContext;
 
-    if (IsChild() || IsRouter())
+#if OPENTHREAD_CONFIG_BORDER_ROUTING_ENABLE
+    IgnoreError(Get<BorderRouter::RoutingManager>().SetEnabled(false));
```

Comment:
`SetEnabled` disables routing manager. It won't recover even after restarting Thread again.

But user would expect Routing Manager **NOT** to be disabled without calling `otBorderRoutingSetEnabled(false)`.

Reply:
+1
sounds `Stops` is a better fit.

Reply:
Good point. Changed in new push. `Stop()` is `private`  so added a new `RequestStop()` for this (with docs saying that it is an immediate stop but `RM` may be started again on role and/or infraif state changes.